src/LtiGrade.php

```diff
@@ -154,8 +154,15 @@ public function getCanvasExtension()
         return $this->canvas_extension;
     }
 
-    // Custom Extension for Canvas.
-    // https://documentation.instructure.com/doc/api/score.html
+    /**
+     * Add custom extensions for Canvas.
+     *
+     * Disclaimer: You should only set this if your LMS is Canvas.
+     *             Some LMS (e.g. Schoology) include validation logic that will throw if there
+     *             is unexpected data. And, the type of LMS cannot simply be inferred by their URL.
+     *
+     * @see https://documentation.instructure.com/doc/api/score.html
+     */
```

Comment:
I don't wanna make backwards-incompatible changes, but I wonder if we should add some sort of method for generically setting custom extensions? Maybe for our next major version of the library?

Reply:
Yeah, this also felt very specific to me. It'd be neat to have something like `setExtension(LmsType $lmsType, array $data): void`.

Reply:
Tangentially related thought that came to mind: it'd typically preferable to do `/** @deprecated */`between for awhile, so I don't think we'd need to introduce a breaking change to make that happen.